common/rfb/util.h

```diff
@@ -51,16 +51,21 @@ namespace rfb {
     CharArray() : buf(0) {}
     CharArray(char* str) : buf(str) {} // note: assumes ownership
     CharArray(size_t len) {
+      length = len;
       buf = new char[len]();
     }
     ~CharArray() {
-      delete [] buf;
+      if (buf) {
+        delete [] buf;
+        buf = nullptr;
+      }
     }
     void format(const char *fmt, ...) __printf_attr(2, 3);
     // Get the buffer pointer & clear it (i.e. caller takes ownership)
     char* takeBuf() {char* tmp = buf; buf = 0; return tmp;}
-    void replaceBuf(char* b) {delete [] buf; buf = b;}
+    void replaceBuf(char* b) {if (buf) delete [] buf; buf = b;}
```

Comment:
What about an update of `length` when `replaceBuf()` is called?

Reply:
Good catch.